### PR TITLE
Introduce ANY_PATTERN_NODE_TYPE and NON_PATTERN_NODE_TYPE

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Introduced ANY_PATTERN_NODE_TYPE and NON_PATTERN_NODE_TYPE. ANY_PATTERN_NODE_TYPE  is a node inside the pattern, that matches on any node type, while NON_PATTERN_NODE_TYPE is outside the pattern node. NON_PATTERN_NODE_TYPE is used to reflect some outgoing edges inside the pattern

### Reason for changes

These changes allow adding the L2Norm pattern.

### Related tickets

67283

### Tests

The current tests were updated with new logic.

### CODE REVIEW

**Review by vshampor — CHANGES_REQUESTED** [review]
As discussed, please try the additional isomorphism search step in the instead of maintaining a dict of node keys vs matches.

File: nncf/common/graph/patterns.py [hunk 1]
```diff
@@ -72,9 +72,11 @@ class GraphPattern:
     Describes layer patterns in model's graph.
     This class is used in quantizer arrangement search algorithm, representing layer fusing patterns
 
-    :param PATTERN_INPUT_NODE_TYPE: Special node type possible pattern input
+    :param ANY_PATTERN_NODE_TYPE: Special node type, meaning any type inside the pattern
+    :param NON_PATTERN_NODE_TYPE: Special node type, meaning any type outside the pattern
     """
-    PATTERN_INPUT_NODE_TYPE = 'INPUT_NODE'
+    ANY_PATTERN_NODE_TYPE = 'ANY_PATTERN_NODE'
```
[the remark this answers is not in the document]
Comment (kshpv) [reply nncf/common/graph/patterns.py:78]: I would stay PATTEN word in the name to not confuse users that this node is inside the pattern

**Review by vshampor — CHANGES_REQUESTED** [review]
Refactor tests to use containers and for-loops and we'll merge this.

**Comment by vshampor** [comment]
Jenkins please retry a build

**Comment by kshpv** [comment]
Jenkins please retry a build

**Review by vshampor — APPROVED** [review]

**Review by andrey-churkin — APPROVED** [review]

**Pull request merged**